internal/controller/resourcesetinputprovider_controller.go

```diff
@@ -522,6 +532,19 @@ func (r *ResourceSetInputProviderReconciler) getGitLabToken(
 	return password, err
 }
 
+// getAzureDevOpsToken returns the appropriate AzureDevOps token by reading the secrets in authData.
+func (r *ResourceSetInputProviderReconciler) getAzureDevOpsToken(
```

Comment:
We should support workload identity as the prefered auth method for Azure. For this you can use this library:

https://github.com/fluxcd/pkg/blob/dcd0517161989dde02817792675b9e6af452f5bf/auth/get_token.go#L33

https://github.com/fluxcd/pkg/blob/dcd0517161989dde02817792675b9e6af452f5bf/auth/azure/scopes.go#L20-L21

`auth.GetToken(ctx, azure.Provider{}, auth.WithScopes(azure.ScopeDevOps))`

Should also introduce a `.spec.serviceAccountName` field in RSIP and pass it to the `auth` options using `auth.WithServiceAccount()` if specified. The field should be optional.

Should also pass the `r.TokenCache` using `auth.WithCache()` like we do in the `getGitHubToken()` method.

Reply:
I would not add workload identity in this PR.